src/modules/ejector/ejector.py

```diff
@@ -0,0 +1,198 @@
+
+        return validators_to_eject
+
+    @lru_cache(maxsize=1)
+    def _get_total_balance(self, blockstamp: BlockStamp) -> Wei:
+        return Wei(
+            self._get_withdrawal_balance(blockstamp) +
+            self._get_el_vault_balance(blockstamp) +
+            self._get_reserved_buffer(blockstamp)
+        )
+
+    def _get_withdrawal_balance(self, blockstamp: BlockStamp) -> Wei:
+        return Wei(self.w3.eth.get_balance(
+            self.w3.lido_contracts.lido_locator.functions.withdrawalVault().call(
+                block_identifier=blockstamp.block_hash
+            ),
+            block_identifier=blockstamp.block_hash,
+        ))
+
+    def _get_el_vault_balance(self, blockstamp: BlockStamp) -> Wei:
+        return Wei(self.w3.eth.get_balance(
+            self.w3.lido_contracts.lido_locator.functions.elRewardsVault().call(
+                block_identifier=blockstamp.block_hash
+            ),
+            block_identifier=blockstamp.block_hash,
+        ))
+
+    def _get_reserved_buffer(self, blockstamp: BlockStamp) -> Wei:
+        return Wei(
+            self.w3.lido_contracts.lido.functions.getBufferedEther().call(
+                block_identifier=blockstamp.block_hash
+            )
+        )
+
+    def get_total_unfinalized_withdrawal_requests_amount(self, blockstamp: BlockStamp) -> Wei:
+        steth_to_finalize = self.w3.lido_contracts.withdrawal_queue_nft.functions.unfinalizedStETH().call(
+            block_identifier=blockstamp.block_hash,
+        )
+        logger.info({'msg': 'Wei to finalize.'})
+        return steth_to_finalize
+
+    def _get_withdrawal_epoch_for_latest_validator(self, blockstamp: BlockStamp, validators_to_eject_count: int) -> EpochNumber:
+        max_exit_epoch_number = 0
+        latest_to_exit_validators_count = 0
+
+        for validator in self.w3.cc.get_validators(blockstamp.state_root):
+            val_exit_epoch = EpochNumber(int(validator.validator.exit_epoch))
+
+            if val_exit_epoch == FAR_FUTURE_EPOCH:
+                continue
+
+            if val_exit_epoch == max_exit_epoch_number:
+                latest_to_exit_validators_count += 1
+
+            elif val_exit_epoch > max_exit_epoch_number:
+                max_exit_epoch_number = val_exit_epoch
+                latest_to_exit_validators_count = 1
+
+        churn_limit = self._get_churn_limit(blockstamp)
+
+        free_slots_in_current_epoch = churn_limit - latest_to_exit_validators_count
+        need_to_exit_all_epochs = (validators_to_eject_count - free_slots_in_current_epoch) // churn_limit + 1
+
+        return EpochNumber(max_exit_epoch_number + need_to_exit_all_epochs + MIN_VALIDATOR_WITHDRAWABILITY_DELAY)
+
+    def _get_sweep_delay_in_epochs(self, blockstamp: BlockStamp):
+        validators = self.w3.cc.get_validators(blockstamp.state_root)
+
+        def if_validators_balance_withdrawable(validator: Validator):
```

Comment:
maybe we can roll back to the spec's definitions to have a good reference for readers?

Reply:
suppose we have a validator with a balance '31 ETH', and '0x01' creds, and `exit_epoch == FAR_FUTURE_EPOCH`
this func would treat this validator as `withdrawable`, while following the genuine spec of CL, this validator has no available skimmable rewards